react_ujs/index.js

```diff
@@ -80,6 +83,18 @@ var ReactRailsUJS = {
     return ReactDOMServer[renderFunction](element)
   },
 
+  // Render `component` using the specified `renderFunction` from `react-dom`.
+  // Override this function to render components in a custom way.
+  // function signature: ("hydrate" | "render", component, node, props)
+  renderComponent: renderWithReactDOM,
+
+  // Enables hot reload for component rendering.
+  //
+  // See the HMR section in README to ensure required steps are completed.
+  useHotReload: function(requireContext) {
+    this.renderComponent = renderWithHotReload(requireContext)
+  },
```

Comment:
Let me know if this comment could be more helpful, i.e. should the steps to setup be inlined here?

Reply:
In addition, there is an opportunity to "stack" instead of "replace". See the example and discussion at https://github.com/reactjs/react-rails/pull/1064. The proposed approach here mirrors `getConstructor` and `useContext`. Let me know if the "stack" option would be a better option